src/textual/app.py

```diff
@@ -92,21 +93,21 @@ class App(Generic[ReturnType], DOMNode):
     def __init__(
         self,
         driver_class: Type[Driver] | None = None,
-        log: str = "",
+        log: str | PurePath = "",
```

Comment:
nothing else to do here, as:
 -  `log` is only used with the builtin function `open()` - which supports PathLike objects out of the box
 - `css_file`is only passed to a `FileMonitor`, which in turn will use it with the `os.stat()` function - which also supports PathLike objects out of the box

Reply:
On reflection it should probably be `log_path` and `css_path`, which I think makes it clear. But I pro the PurePath for typing purposes.

Reply:
I agree that `_path`  suffixes should make the nature of these params clearer! :slightly_smiling_face: 

I started to rename them, however I ran into an unexpected issue: doing `self.css_path = css_path` in the App constructor is not possible, because it's a subclass of DOMNode and this one already has a `css_path` read-only property :pensive: 
This `DOMNode.css_path` property seems to be used only once in the Textual codebase, so maybe we can rename it to something like `nodes_css_path` for example, as it returns a `list[DOMNode]` ?

Reply:
Good point. Suggest `css_path_nodes`